Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -98,7 +98,7 @@ def fix_dynamic_linkage
 
   def loader_name_for(file, target)
     # Use @loader_path-relative install names for other Homebrew-installed binaries.
-    if ENV["HOMEBREW_RELOCATABLE_INSTALL_NAMES"] && target.start_with?(HOMEBREW_PREFIX)
+    if !ENV["HOMEBREW_NO_RELOCATABLE_INSTALL_NAMES"] && target.start_with?(HOMEBREW_PREFIX)
```

Comment:
```suggestion
    if target.start_with?(HOMEBREW_PREFIX)
```

makes sense to me if this has been disabled by default and working for that long